packages/client/hmi-client/src/workflow/ops/model-config/tera-model-config.vue

```diff
@@ -132,27 +132,35 @@ const updateState = (updatedField) => {
 	emit('update-state', state);
 };
 
-const updateConfigParam = (param: ModelParameter) => {
+const updateConfigParam = (params: ModelParameter[], updateAll = false) => {
 	const state = _.cloneDeep(props.node.state);
 	if (!state.parameters) return;
 
-	// find param with the same id and update it
-	const idx = state.parameters.findIndex((p) => p.id === param.id);
-	if (idx !== -1) {
-		state.parameters[idx] = param;
+	if (updateAll) {
+		state.parameters = params;
+	} else {
+		// find param with the same id and update it
+		const idx = state.parameters.findIndex((p) => p.id === params[0].id);
+		if (idx !== -1) {
+			state.parameters[idx] = params[0];
+		}
```

Comment:
The update functions have confusing signatures - I had to dive deep into the implementation to understand how the array and boolean flag interact with each other. 

We can leave it alone for now, but let's have a chat when you get around to clean things up.

Reply:
I did initially think about having separate functions for updating (e.g. updateSingleParam and updateAllParams or something), but thought this approach miigghttt be okay LOL. But happy to make changes if necessary.